arbnode/dataposter/redis_storage.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package dataposter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/go-redis/redis/v8"
+)
+
+// Requires that Item is RLP encodable/decodable
+type RedisStorage[Item any] struct {
+	client redis.UniversalClient
+	key    string
+}
+
+func NewRedisStorage[Item any](redisUrl string, key string) (*RedisStorage[Item], error) {
+	redisOptions, err := redis.ParseURL(redisUrl)
+	if err != nil {
+		return nil, err
+	}
+	client := redis.NewClient(redisOptions)
+	return &RedisStorage[Item]{client, key}, nil
+}
+
+func (s *RedisStorage[Item]) GetContents(ctx context.Context, startingIndex uint64, maxResults uint64) ([]*Item, error) {
+	query := redis.ZRangeArgs{
+		Key:     s.key,
+		ByScore: true,
+		Start:   int64(startingIndex),
+		Count:   int64(maxResults),
+	}
+	itemStrings, err := s.client.ZRangeArgs(ctx, query).Result()
+	if err != nil {
+		return nil, err
+	}
+	var items []*Item
+	for _, itemString := range itemStrings {
+		var item Item
+		err := rlp.DecodeBytes([]byte(itemString), &item)
+		if err != nil {
+			return nil, err
+		}
+		items = append(items, &item)
+	}
+	return items, nil
+}
+
+func (s *RedisStorage[Item]) GetLast(ctx context.Context) (*Item, error) {
+	query := redis.ZRangeArgs{
```

Comment:
I'd do something like: 
{"Stop: "+inf", Rev: true, Count: 1}
Yours seems much better then mine, and I trust that it works.. just didn't yet find a redis documentation that says it does. Have a pointer?

Reply:
oh, got it! resolving

Reply:
Yeah, the docs are terrible on this front. There are so many ways to list items in a sorted set and they're all terrible